packages/forklift-console-plugin/src/modules/Providers/views/details/components/DetailsSection/components/UIDetailsItem.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { useModal } from 'src/modules/Providers/modals';
+import { EditProviderUIModal } from 'src/modules/Providers/modals/EditProviderUI';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import { ExternalLink } from '@kubev2v/common';
+import { V1beta1Provider } from '@kubev2v/types';
+
+import { DetailsItem } from '../../../../../utils';
+
+import { ProviderDetailsItemProps } from './ProviderDetailsItem';
+
+/**
+ * @typedef {Object} UIDetailsItemProps - extends ProviderDetailsItemProps
+
+ * @property {string} [text] - A text to be displayed as a content.
+ * @property {function} [calcProviderUI] - A method for auto calculating the provider's UI path.
+ **/
+export interface UIDetailsItemProps extends ProviderDetailsItemProps {
+  text: string;
+  calcProviderUI: (provider: V1beta1Provider) => string;
+}
+
+const getProviderUIAnnotation = (provider: V1beta1Provider): string =>
+  provider?.metadata?.annotations?.['forklift.konveyor.io/providerUI'];
+
+export const UIDetailsItem: React.FC<UIDetailsItemProps> = ({
+  resource: provider,
+  canPatch,
+  text,
+  calcProviderUI,
+}) => {
+  const { t } = useForkliftTranslation();
+  const { showModal } = useModal();
+  // Read the annotation and only if empty, try to auto calculate the provider UI path
+  const content =
+    getProviderUIAnnotation(provider) ?? (calcProviderUI ? calcProviderUI(provider) : null);
+
+  return (
+    <DetailsItem
```

Comment:
```suggestion
    <EditableContentButton
```

Reply:
I prefer the current implementation of wrapping that with the `DetailsItem` for 2 reasons:
1. it's much more flexibility for  future usages. We might re-use this with a title, disable the editable option etc. 
Current implementation enable all options for items within the details page.

2. Additionally, if the user doesn't have patching permissions, then it will be `NonEditableContent` and not `EditableContentButton`
So we can move this logic to `UIDetailsItem` but why duplicating that?

Reply:
My concern it that it is a content not an item, I don't want to use the name "Item" for a content

> t's much more flexibility 

It is a content, it is flexibel enogh

> if the user doesn't have patching permissions,

That is a good point, I prefer to address this concern by enhancing  `EditableContentButton` so it will render `NonEditableContent` when `onEdit` is falsy, that way it is still a content item, but it have more general use case, it will also simplify  `DetailsItem`

Reply:
The `UIDetailsItem` was replaced with adding the web ui link into the `NameAndWebUiLinkDetailsItem` comp.
So this is not relevant anymore.